### ISSUE TITLE
I can't achieve the accuracy in bench mark, could somebody help?

### ISSUE THREAD

**Issue by GranMin** [peteryuX/arcface-tf2#38]
![output19 42](https://user-images.githubusercontent.com/39046019/129169882-c73b3944-fd3f-43cc-8e73-46407d81848c.png)
I use the same train dataset and test dataset as you proposed, but the best result I've got so far is as the picture shows.
I used SGD optimizer and lr=0.1,0.05,0.01,0.0001,0.00001, each lr an epoch. And when I found the loss increasing rather than decreasing, I stoped training. And I got the test result for loss 19.42 as up picture.
More, this is test result when the train loss is 21.15, shown as down picture.
![output21 15](https://user-images.githubusercontent.com/39046019/129170796-f8ea740b-5e96-4ebd-bc26-efa5a4bd1206.png)

**Comment by Androsimus** [comment]
@GranMin what backbone do you use?
I tried MobileNetV2 and got similar to author's results.
I used constant learning rate = 0.01 during about 10 epochs and achieved loss about 10.
I think you should try similar to my lr schedule at first and decrease lr only after that (and maybe not so fast). Otherwise your model haven't enough time to use relatively big gradients to decrease loss.

**Comment by GranMin** [comment]
@Androsimus I used the Resnet50. I will try and reply as soon. Thanks for your advice.

**Comment by GranMin** [comment]
the newest info: I use the Resnet50, SGD optimizer and set lr=0.01, after 30 epoch, I reached 99.08 in lfw, and 92 in AgeDB-30, as shown follow.
![image](https://user-images.githubusercontent.com/39046019/129830767-0dc8a2f1-105d-40ef-8d61-9901edce1a47.png)

**Comment by Androsimus** [comment]
@GranMin Maybe use of 30 epochs is too much and you have got some overfitting?.. 
Do you have previous checkpoints from 10-15 epochs? If yes, try to check them on val datasets.

**Comment by GranMin** [comment]
@Androsimus I have the same feeling of too much epochs.But the loss of 10-15 epoch is about 20, and the acc of lfw is about 98.

**Comment by Androsimus** [comment]
@GranMin This is very strange. Maybe you changed some other parameters? Maybe parameters of Arcface: margin, scale?
Because there are other issues, where persons wrote about good results using Resnet50 on native for this repository dataset.

**Comment by GranMin** [comment]
@Androsimus I don't change any other parameters. And I tried NormHead for one epoch, then use the Archead, it's amazing that just after one epoch in Archead, the loss comes to about 11. But then the same phenomenon took place: the loss increase a few at the begin of the epoch, and then decrease, but at very low speed. Like this:
![image](https://user-images.githubusercontent.com/39046019/130022727-f4f11606-0ad6-404d-86cd-a0fdb40fc7ec.png)

**Comment by Androsimus** [comment]
@GranMin I'm not sure how NormHead is supposed to use. Maybe as a warmup.
But the NormHead and the ArcHead are completely different. As far as I understand, NormHead is for ordinary classification, if so, then classification problem is much easier and due to this you achieve low loss much faster. Otherwise, when you use ArcHead with its margin and scale parameters and its different concept the classification problem is harder.
But I don't understand your situation: on the one hand you said about loss ~20 on 10-15 epochs, on the other hand after first epoch you had loss <11 and then it decreased at low speed...

To sum up. I used strictly ArcHead, I suppose other people did it too. So I propose to try using only ArcHead.

**Comment by GranMin** [comment]
@Androsimus In fact, I tried two times about training the model recently.
The first time, I use only ArcHead and train at lr=0.01 with SGD for 10-15 epochs, loss ~20. Finally, I trained about 30 epochs to get a loss ~6.5 and accuracy 99.06 on lfw.
The second time, I tried the NormHead for one epoch and the changed to Archead, after one epoch with ArcHead, loss down to 10. But as described latest, the speed come down.
As for the difference of two head in math is that NormHead just use softmax to ensure correct classification, it work not so well on boundary between classes. And Archead forces a theta between two classes, to avoid two classes adjoin with each other.

**Comment by Androsimus** [comment]
@GranMin 
If you look at this post and https://github.com/peteryuX/arcface-tf2/issues/4#issuecomment-599015569 and thread, then you will see big differences from your results. Strange.
Nevertheless, if you have your old logs and chechpoints from training using only ArcHead, then try to find checkpoint that correspond to loss about 8-9 and try it on validation datasets.

**Comment by GranMin** [comment]
results as:
first for loss 9.16 and second for loss 8.00
![1](https://user-images.githubusercontent.com/39046019/130043578-147e7d9e-ef6e-4c38-8ed9-2ba30e18eff8.png)
![2](https://user-images.githubusercontent.com/39046019/130043601-30ff910e-0970-44ab-bfae-793bd12eddda.png)

**Comment by Androsimus** [comment]
@GranMin this is some mystery )

**Comment by Androsimus** [comment]
@GranMin could you post your config file *.yaml?
Anyway if you figure out the reason of that strange model training behavior, please write about it.

**Comment by GranMin** [comment]
@Androsimus Sorry for long waiting. Hahah...I just took a vacation to Jiuzhai Gou nature reserve last week.
I talk to my teacher and then I know they broaden my dataset with some asian faces.
I redown the dataset the author provided, and got this result just for 2 epoch, just use arcface head.
![image](https://user-images.githubusercontent.com/39046019/131282309-795d654a-e9c4-434c-ad09-9b55c21e16f2.png)
I decide to train resnet152 from scratch following time. And the experience that use softmax first may also help.
Again, thank you for your advice~best wishes, my friend!

**Comment by GranMin** [comment]
And for 5 epoch finished, this is final result:
![image](https://user-images.githubusercontent.com/39046019/131310901-23d5f6e0-0e4a-406e-906c-cbdf939fbdf0.png)

**Comment by Androsimus** [comment]
@GranMin glad you got nice results :)
Best wishes!

**Comment by rahul1201dtu** [comment]
> the newest info: I use the Resnet50, SGD optimizer and set lr=0.01, after 30 epoch, I reached 99.08 in lfw, and 92 in AgeDB-30, as shown follow. ![image](https://user-images.githubusercontent.com/39046019/129830767-0dc8a2f1-105d-40ef-8d61-9901edce1a47.png)

hello can you please tell model

> I use the same train dataset and test dataset as you proposed, but the best result I've got so far is as the picture shows.
> I used SGD optimizer and lr=0.1,0.05,0.01,0.0001,0.00001, each lr an epoch. And when I found the loss increasing rather than decreasing, I stoped training. And I got the test result for loss 19.42 as up picture.
> More, this is test result when the train loss is 21.15, shown as down picture.

can you please tell, me if you used the same architecture  as mentioned in model.py or a different  because I used the  Resnet50 with few modifications in dense layer as and during my  2nd  epoch my training loss is 17.85 so I just want to know whether I am doing   fine or not and after how many epochs will i get test accuracy of 99% and my code is below




def initialize_base_network():
    target_size = (112,112)
    xecption_model=tf.keras.applications.ResNet50(weights='imagenet', input_shape = target_size + (3,), include_top=False)
    #global_layer = layers.Flatten()( xecption_model.output)
    for layer in  xecption_model.layers:
        layer.trainable = True
    x = layers.BatchNormalization()(xecption_model.output)
    x = Dropout(0.2)(x)
    x = Flatten()(x)
    x = Dense(512, use_bias=True, kernel_initializer='he_normal',kernel_regularizer=regularizers.l2(weight_decay))(x)

    x = BatchNormalization(momentum=0.9, scale=False, epsilon=2e-5, name='embedding')(x)


        # x = BatchNormalization()(x)
        # x = Dropout(rate=0.5)(x)
        # x = Flatten()(x)
        # x = Dense(embd_shape, kernel_regularizer=_regularizer(w_decay))(x)
    #     # x = BatchNormalization()(x)

    # x = layers.Dense(512,activation='linear')(global_layer)
    # x = layers.BatchNormalization()(x)


    return Model(inputs = xecption_model.inputs, outputs = x)

**Comment by xalbertoisorna** [comment]
> @Androsimus In fact, I tried two times about training the model recently. The first time, I use only ArcHead and train at lr=0.01 with SGD for 10-15 epochs, loss ~20. Finally, I trained about 30 epochs to get a loss ~6.5 and accuracy 99.06 on lfw. The second time, I tried the NormHead for one epoch and the changed to Archead, after one epoch with ArcHead, loss down to 10. But as described latest, the speed come down. As for the difference of two head in math is that NormHead just use softmax to ensure correct classification, it work not so well on boundary between classes. And Archead forces a theta between two classes, to avoid two classes adjoin with each other.

How do you change head (normhead to archead)? I tried it but I have this error:
`    raise ValueError(
ValueError: Cannot assign value to variable ' conv2d/bias:0': Shape mismatch.The variable shape (24,), and the assigned value shape (32,) are incompatible.
`
